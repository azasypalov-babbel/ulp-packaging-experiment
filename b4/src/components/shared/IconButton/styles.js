import styled from 'styled-components';
import { Button } from '../Button/Button';
import { sizeStyles, paddingWithBorderStyles } from '../Button/commonButtonStyles';
import primaryButtonStyles from '../Button/primaryButtonStyles';

export const buttonBorderWidth = '1px';

export const IconWrapper = styled.div`
  display: flex;
  justify-content: center;
`;

export const ContentWrapper = styled.span`
  flex: 1 1 100%;
  margin-left: 0.5rem;
`;


export const StyledIconButton = styled(Button)`
  ${paddingWithBorderStyles('1px', '0.75rem')}

  ${sizeStyles}

  min-width: 2.5rem;

  color: ${(props) => props.theme.cascada.storm};

  &:active {
    color: ${(props) => props.theme.cascada.white};
  }

  ${({ primary }) => primary ? primaryButtonStyles : null}

  @media (min-width: ${(props) => props.theme.viewports.breakpoints.xsmall}) {
    min-width: 3rem;
  }
`;

export const StyledIconButtonContent = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;
