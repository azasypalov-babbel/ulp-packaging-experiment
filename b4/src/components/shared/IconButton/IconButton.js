import React from 'react';
import PropTypes from 'prop-types';
import * as icons from '../icons';
import { StyledIconButton, IconWrapper, ContentWrapper, StyledIconButtonContent } from './styles';
import withKeypress from '../withKeypress';

export const IconButton = ({
  children,
  keyboardHintComponent,
  iconName,
  contentClassName,
  onClick,
  ...props
}) => {
  const Icon = icons[iconName];

  const handleClick = (event) => {
    event.target.blur();
    onClick(event);
  };

  return (
    <StyledIconButton {...props} onClick={handleClick}>
      <StyledIconButtonContent>
        <IconWrapper>
          {keyboardHintComponent ||
            <Icon
              size={'1.5rem'}
              fillColor="currentColor"
            />
          }
        </IconWrapper>
        {children && <ContentWrapper className={contentClassName}>
          {children}
        </ContentWrapper>}
      </StyledIconButtonContent>
    </StyledIconButton>
  );
};

IconButton.displayName = `IconButton`;

IconButton.propTypes = {
  keyboardHintComponent: PropTypes.node,
  iconName: PropTypes.oneOf([
    'KeyboardIcon',
    'CheckIcon',
    'CrossIcon',
    'MicIcon',
    'MicOff',
    'ShowIcon',
    'LightBulbIcon',
    'HideIcon',
    'BackspaceIcon'
  ]).isRequired,
  children: PropTypes.node,
  contentClassName: PropTypes.string,
  onClick: PropTypes.func
};


IconButton.defaultProps = {
  type: 'button'
};

export default withKeypress(IconButton);
