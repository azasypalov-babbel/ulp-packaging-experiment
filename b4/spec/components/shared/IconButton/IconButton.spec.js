import React from 'react';
import { IconButton } from '../../../../src/components/shared/IconButton/IconButton';
import { shallow } from 'enzyme';

describe('<IconButton />', () => {
  describe('<IconButton positive iconName="CheckIcon" />', () => {
    test('it renders', () => {
      const wrapper = shallow(<IconButton positive iconName="CheckIcon"/>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<IconButton negative iconName="CrossIcon" />', () => {
    test('it renders', () => {
      const wrapper = shallow(<IconButton negative iconName="CrossIcon" />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<IconButton iconName="KeyboardIcon" />', () => {
    test('it renders', () => {
      const wrapper = shallow(<IconButton iconName="KeyboardIcon"/>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<IconButton /> with keyboardHintComponent', () => {
    const KeyboardHintComponent = () => <span>Hint</span>;
    test('it renders', () => {
      const wrapper = shallow(<IconButton
        iconName="KeyboardIcon"
        keyboardHintComponent={<KeyboardHintComponent />}
      />);

      expect(wrapper.find(KeyboardHintComponent)).toHaveLength(1);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
