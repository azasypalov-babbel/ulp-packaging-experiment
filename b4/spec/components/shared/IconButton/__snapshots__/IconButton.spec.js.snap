// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> <IconButton /> with keyboardHintComponent it renders 1`] = `
<Styled(styled.button)
  onClick={[Function]}
  type="button"
>
  <styled.div>
    <styled.div>
      <KeyboardHintComponent />
    </styled.div>
  </styled.div>
</Styled(styled.button)>
`;

exports[`<IconButton /> <IconButton iconName="KeyboardIcon" /> it renders 1`] = `
<Styled(styled.button)
  onClick={[Function]}
  type="button"
>
  <styled.div>
    <styled.div>
      <WithTheme(KeyboardIcon)
        fillColor="currentColor"
        size="1.5rem"
      />
    </styled.div>
  </styled.div>
</Styled(styled.button)>
`;

exports[`<IconButton /> <IconButton negative iconName="CrossIcon" /> it renders 1`] = `
<Styled(styled.button)
  negative={true}
  onClick={[Function]}
  type="button"
>
  <styled.div>
    <styled.div>
      <WithTheme(CrossIcon)
        fillColor="currentColor"
        size="1.5rem"
      />
    </styled.div>
  </styled.div>
</Styled(styled.button)>
`;

exports[`<IconButton /> <IconButton positive iconName="CheckIcon" /> it renders 1`] = `
<Styled(styled.button)
  onClick={[Function]}
  positive={true}
  type="button"
>
  <styled.div>
    <styled.div>
      <WithTheme(CheckIcon)
        fillColor="currentColor"
        size="1.5rem"
      />
    </styled.div>
  </styled.div>
</Styled(styled.button)>
`;
